api/specs.proto

```diff
@@ -258,6 +258,12 @@ message ContainerSpec {
 	// task will exit and a new task will be rescheduled elsewhere. A container
 	// is considered unhealthy after `Retries` number of consecutive failures.
 	HealthConfig healthcheck = 16;
+
+	message SecurityConfig {
+	    string userns = 1;
+	    string credential_spec = 2 [(gogoproto.customname) = "CredentialSpec"];
```

Comment:
I think maybe that particular objection was to the `docker service create` CLI flags.  (cc @diogomonica)

AFAIK the current plan is for low level capabilities to go into the spec and the API.  Possibly in a versioned object (or typed objects, like "custom" for the low level capabilities), and for the something else (like the CLI) to translate from high level grants to groups low level capabilities.  This is the current tentative plan for security profiles, because it's likely we won't get the first iteration of high level profiles right.  

Even if the CLI is not the correct place to put the mapping, and the API needs to accept the higher level API instead, having a version of the profile that contains the low level capabilities in case someone wanted to specify a custom profile could be useful.

But to reiterate @stevvooe's question - I wasn't sure how the other security-related options would relate to security profiles in this object model (should profiles be a separate object in a security configuration for instance, etc.)